modules/ding_webtrekk/ding_webtrekk.js

```diff
@@ -0,0 +1,131 @@
+      });
+
+      // Secial handling for ding entity rating event.
+      //
+      // It would require a complete rework of the ding_entity_rating_display
+      // theme hook to avoid this. It doesn't provide the ability to add custom
+      // attributes/classes in preprocess and is not using using render arrays
+      // for rating elements, making them problematic to modify.
+      $('.ding-webtrekk-rating-event', context).once('ding-webtrekk', function() {
+        var contentId = $(this).data('ding-entity-rating-id');
+        $('.js-rating-symbol', this).each(function(index) {
+          var rating = (index + 1) + '';
+          $(this).click(function() {
+            wts.push(['send', 'click', {
+              linkId:'Materiale rating',
+              customClickParameter: {
+                58: rating,
+                57: contentId
+              }
+            }]);
+          });
+        })
+      });
+
+      // Track autocomplete selections.
+      $('.form-item-search-block-form .form-autocomplete', context)
+        .once('ding-webtrekk')
+        .on('autocompleteSelect', function(e, selected) {
+          if (($(selected).text())) {
+            wts.push(['send', 'click', {
+              linkId: 'Autocomplete søgning clicks',
+              customClickParameter: { 54: $(selected).text() }
+            }]);
+          }
+      });
+
+      // Track ding loan renew selected events.
+      //
+      // The renew all button is attached completely on the server, but for
+      // renew selected, we need information about selected loans in the UI,
+      // before we send event.
+      $('.ding-webtrekk-event-renew-selected', context)
+        .once('ding-webtrekk')
+        .click(function(e) {
+          var selectedMaterials = [];
+          $('.material-item input[type=checkbox]:checked').each(function() {
+            // We have collected the material ids (pids) in our event data
+            // attribute for each select box.
+            selectedMaterials.push($(this).data('ding-webtrekk-event'));
+          });
+          wts.push(['send', 'click', {
+            linkId: 'Forny valgte materialer',
+            customClickParameter: { 55: selectedMaterials.join(';') }
+          }]);
+      });
+
+      // Special handling for ding_carousel.
+      //
+      // The primary reason for this is that we need the title of the carousel
+      // to pass as value in event and URL-parameter. For tabbed carousels the
+      // title of the individuals carousels in the tabs are set in the
+      // data-title attribute of the carousels. For single carousels the title
+      // is not avialable here, but instead it's controlled by the configuration
+      // on the panel pane. There are no panel hooks where we can get the title
+      // and easily change the content tree to add event and URL-parameters to
+      // elements (content is already rendered when hook is run).
+      $('.ding-carousel').each(function() {
```

Comment:
I actually left out context and once here on purpose, since the items in the carousels can be lazy-loaded when the users reach the end of the carousel. I wanted to make sure the selectors applied when new items is added.

I must admit I havn't been able to test if this is necessary, since there was some problems with the lazy-loadings. (probably because of: https://platform.dandigbib.org/issues/3903).

I didn't see it as a big issue though, since all the operations inside the selector are once'd and using context.

I see now that maybe a comment about it would be appropriate?